hack/lib/util.sh

```diff
@@ -798,17 +798,38 @@ function kube::util::md5() {
 
 # kube::util::read-array
 # Reads in stdin and adds it line by line to the array provided. This can be
-# used instead of "mapfile -t", and is bash 3 compatible.
+# used instead of "mapfile -t", and is bash 3 compatible.  If the named array
+# exists and is an array, it will be used.  Otherwise it will be unset and
+# recreated.
 #
 # Assumed vars:
 #   $1 (name of array to create/modify)
 #
 # Example usage:
-# kube::util::read-array files < <(ls -1)
+#   kube::util::read-array files < <(ls -1)
 #
+# When in doubt:
+#   $ X=abc
+#   $ Y=(a b c)
+#   $ unset Z
+#   $ declare -p X Y Z
+#   declare -- X="abc"
+#   declare -a Y=([0]="a" [1]="b" [2]="c")
+#   bash: declare: Z: not found
+#   $ kube::util::read-array X < <(echo -ne "1 1\n2 2\n3 3\n")
+#   $ kube::util::read-array Y < <(echo -ne "1 1\n2 2\n3 3\n")
+#   $ kube::util::read-array Z < <(echo -ne "1 1\n2 2\n3 3\n")
+#   $ declare -p X Y Z
+#   declare -a X=([0]="1 1" [1]="2 2" [2]="3 3")
+#   declare -a Y=([0]="1 1" [1]="2 2" [2]="3 3")
+#   declare -a Z=([0]="1 1" [1]="2 2" [2]="3 3")
```

Comment:
It's... hard to imagine a language with a less user friendly take on arrays. (just complaining, not asking for a change, unless you can fix bash)

Reply:
We could replace all this with python or go in a lter iteration